### PR TITLE
Fix input getting problem by setting content type

### LINKED ISSUE #345: Campaign > http stage :  body input is not passed to http request
When using an Http stage as part of a campaign, if the body is specified as one of the inputs, its value is not then passed to the http provider.  This means that the http request is sent without containing the properties passed. 

An example of this is when creating a POST to LogicApp as part of an approval stage in a campaign. Information such as instance, solution and target is not sent to the LogicApp be included as part of the email.

image

Version Seen: [0.48.25](https://github.com/eclipse-symphony/symphony/tree/0.48.25)

### PR DESCRIPTION
**Pull request by lirenjie95**
fixed #345

### CODE REVIEW

File: api/pkg/apis/v1alpha1/providers/stage/http/http.go [hunk 1]
```diff
@@ -214,6 +214,7 @@ func (i *HttpStageProvider) Process(ctx context.Context, mgrContext contexts.Man
 				return nil, false, err
 			}
 			req.Body = ioutil.NopCloser(bytes.NewBuffer(jData))
+			req.Header.Set("Content-Type", "application/json; charset=UTF-8")
```
Comment (LeiQL) [inline api/pkg/apis/v1alpha1/providers/stage/http/http.go:217]: Is this set as configurable input?

Comment (lirenjie95) [reply api/pkg/apis/v1alpha1/providers/stage/http/http.go:217]: In the future, we might consider to support XML/other types.

**Review by RemindD — APPROVED** [review]
Good catch.

File: api/pkg/apis/v1alpha1/providers/stage/http/http_test.go [hunk 2]
```diff
@@ -73,6 +74,35 @@ func TestPingBing(t *testing.T) {
 	assert.Equal(t, 200, outputs["status"])
 	assert.NotNil(t, outputs["body"])
 }
+func TestPostRequestWithJson(t *testing.T) {
+	provider := HttpStageProvider{}
+	err := provider.Init(HttpStageProviderConfig{
+		Method: "POST",
+		Url:    "https://jsonplaceholder.typicode.com/posts",
+	})
```
Comment (iwangjintian) [inline api/pkg/apis/v1alpha1/providers/stage/http/http_test.go:82]: what's this url?

Comment (lirenjie95) [reply api/pkg/apis/v1alpha1/providers/stage/http/http_test.go:82]: It's a POST request testing mock website

**Review by Haishi2016 — APPROVED** [review]

**Pull request merged**